lib/Models/MapboxVectorTileCatalogItem.ts

```diff
@@ -0,0 +1,152 @@
+
+  constructor(readonly item: MapboxVectorTileCatalogItem) {
+    super();
+  }
+
+  duplicateLoadableStratum(newModel: BaseModel): this {
+    return new MapboxVectorTileLoadableStratum(
+      newModel as MapboxVectorTileCatalogItem
+    ) as this;
+  }
+
+  static async load(item: MapboxVectorTileCatalogItem) {
+    return new MapboxVectorTileLoadableStratum(item);
+  }
+
+  @computed get legends() {
+    return [
+      createStratumInstance(LegendTraits, {
+        items: [
+          createStratumInstance(LegendItemTraits, {
+            color: this.item.fillColor,
+            outlineColor: this.item.lineColor,
+            title: this.item.name
+          })
+        ]
+      })
+    ];
+  }
+}
+
+StratumOrder.addLoadStratum(MapboxVectorTileLoadableStratum.stratumName);
+
+class MapboxVectorTileCatalogItem extends AsyncMappableMixin(
+  UrlMixin(CatalogMemberMixin(CreateModel(MapboxVectorTileCatalogItemTraits)))
+) {
+  @observable
+  private imageryProvider: ImageryProvider | undefined;
+
+  public readonly forceProxy = true;
+
+  static readonly type = "mvt";
+
+  get type() {
+    return MapboxVectorTileCatalogItem.type;
+  }
+
+  get typeName() {
+    return i18next.t("models.mapboxVectorTile.name");
+  }
+  readonly canZoomTo = true;
+
+  async forceLoadMetadata() {
+    const stratum = await MapboxVectorTileLoadableStratum.load(this);
+    runInAction(() => {
+      this.strata.set(MapboxVectorTileLoadableStratum.stratumName, stratum);
+    });
+  }
+
+  protected async forceLoadMapItems(): Promise<void> {
+    await this.loadMetadata();
+    this.updateImageryProvider();
+  }
+
+  private updateImageryProvider() {
+    autorun(() => {
+      if (
+        !isDefined(this.url) ||
+        !isDefined(this.layer) ||
+        !isDefined(this.rectangle)
+      ) {
+        return;
+      }
+
+      this.imageryProvider = new MapboxVectorTileImageryProvider({
+        url: this.url,
+        layerName: this.layer,
+        styleFunc: action(() => ({
+          fillStyle: this.fillColor,
+          strokeStyle: this.lineColor,
+          lineJoin: "miter",
+          lineWidth: 1
+        })),
+        rectangle: Rectangle.fromDegrees(
+          this.rectangle.west,
+          this.rectangle.south,
+          this.rectangle.east,
+          this.rectangle.north
+        ),
+        minimumZoom: this.minimumZoom,
+        maximumNativeZoom: this.maximumNativeZoom,
+        maximumZoom: this.maximumZoom,
+        uniqueIdProp: this.idProperty,
+        featureInfoFunc: this.featureInfoFromFeature
+      });
+    });
+  }
+
+  @computed
+  get mapItems() {
+    if (this.isLoadingMapItems || this.imageryProvider === undefined) {
+      return [];
+    }
+
+    return [
+      {
+        imageryProvider: this.imageryProvider,
+        show: this.show,
+        alpha: this.opacity,
+        canZoomTo: this.canZoomTo
+      }
+    ];
+  }
+
+  @action.bound
+  featureInfoFromFeature(feature: VectorTileFeature) {
```

Comment:
I don't think this will be called. `featureInfoFromFeature` is a `TableMixin` function and `TableMixin` isn't being used here.

Reply:
The functionality here should either be ported to MVT Imagery Provider or deleted.

Reply:
That function is used internally here - https://github.com/TerriaJS/terriajs/blob/mobx-vector-imagery-provider-ts-flat/lib/Models/MapboxVectorTileCatalogItem.ts#L110

Reply:
Ahh ok. I missed that it's passed into the ImageryProvider.